health/health.go

```diff
@@ -0,0 +1,35 @@
+package health
+
+import (
+	"errors"
+)
+
+var Reporter = &reporter{
```

Comment:
Why have it on the global state? it might cause race calling it from multiple routines

Reply:
I want the reporter to be a singleton.
On one side, every package can consume the reporter and report an health change.
On the other side, the launcher always listens to that reporter and updates the fleet accordingly.
I will also add it as a comment above that section.

Reply:
I think it's correct to make the reporter a singleton, however still worth protecting the global var from "race conditions".

Reply:
Singleton does not mean it will be on the global state. 
Can we add a test that shows your suggestion @amirbenun that is also concurrency safe?